logger/logger.go

```diff
@@ -23,35 +22,35 @@ type Config struct {
 	FileMaxAgeInDays int
 }
 
-var (
-	L    *slog.Logger
-	once = sync.Once{}
-)
+var l *slog.Logger
 
 func init() {
-
-	fileWriter := addSync(&lumberjack.Logger{
+	fileWriter := &lumberjack.Logger{
 		Filename:  defaultFilePath,
 		LocalTime: defaultUseLocalTime,
 		MaxSize:   defaultFileMaxSizeInMB,
 		MaxAge:    defaultFileAgeInDays,
-	})
-	L = slog.New(
+	}
+	l = slog.New(
 		slog.NewJSONHandler(io.MultiWriter(fileWriter, os.Stdout), &slog.HandlerOptions{}),
 	)
 }
 
-func New(cfg Config, opt *slog.HandlerOptions) {
-	once.Do(func() {
-		fileWriter := addSync(&lumberjack.Logger{
-			Filename:  cfg.FilePath,
-			LocalTime: cfg.UseLocalTime,
-			MaxSize:   cfg.FileMaxSizeInMB,
-			MaxAge:    cfg.FileMaxAgeInDays,
-		})
-
-		L = slog.New(
-			slog.NewJSONHandler(io.MultiWriter(fileWriter, os.Stdout), opt),
-		)
-	})
+func L() *slog.Logger {
+	return l
+}
+
+func New(cfg Config, opt *slog.HandlerOptions) *slog.Logger {
+	fileWriter := &lumberjack.Logger{
+		Filename:  cfg.FilePath,
+		LocalTime: cfg.UseLocalTime,
+		MaxSize:   cfg.FileMaxSizeInMB,
+		MaxAge:    cfg.FileMaxAgeInDays,
+	}
+
+	l = slog.New(
```

Comment:
this `l` is global `l`, use `l := slog.New` to instantiate new variable.

Reply:
Done